### PR TITLE
[sparkle] - enh(Dropdown): add `NewDropdownMenuSearchbar`

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR aims at introducing a new `NewDropdownMenuSearchbar` component to enable search on dropdown items

## Risk

Low

## Deploy Plan

Publish `sparkle`

### CODE REVIEW

File: sparkle/src/components/NewDropdown.tsx [hunk 1]
```diff
@@ -339,6 +340,58 @@ const NewDropdownMenuShortcut = ({
 };
 NewDropdownMenuShortcut.displayName = "NewDropdownMenuShortcut";
 
+interface NewDropdownMenuSearchbarProps {
```
Comment (aubin-tchoi) [inline sparkle/src/components/NewDropdown.tsx:343]: IMHO would be clearer written with an `extends SearchbarProps` + `Omit` if this makes sense, wdyt?

File: sparkle/src/components/NewDropdown.tsx [hunk 2]
```diff
@@ -339,6 +340,52 @@ const NewDropdownMenuShortcut = ({
 };
 NewDropdownMenuShortcut.displayName = "NewDropdownMenuShortcut";
 
+interface NewDropdownMenuSearchbarProps
+  extends Omit<SearchbarProps, "className"> {
```
[the remark this answers is not in the document]
Comment (JulesBelveze) [reply sparkle/src/components/NewDropdown.tsx]: I would actually not omit anything

[the remark this answers is not in the document]
Comment (aubin-tchoi) [reply sparkle/src/components/NewDropdown.tsx]: oh that's right i misread the `handleKeyDown`

**Review by aubin-tchoi — APPROVED** [review]
LGTM

**Pull request merged**